app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -40,17 +40,17 @@ class WallpaperManager(private val settings: Settings) {
     private fun adjustTheme(context: Context) {
         val mode = if (currentWallpaper != Wallpaper.NONE) {
             if (currentWallpaper.isDark) {
-                updateThemePreference(useDarkTheme = true)
                 AppCompatDelegate.MODE_NIGHT_YES
             } else {
-                updateThemePreference(useLightTheme = true)
                 AppCompatDelegate.MODE_NIGHT_NO
             }
         } else {
-            // For the default wallpaper, there is not need to adjust the theme,
-            // as we want to allow users decide which theme they want to have.
-            // The default wallpaper adapts to whichever theme the user has.
-            return
+            when {
+                settings.shouldUseLightTheme -> AppCompatDelegate.MODE_NIGHT_NO
+                settings.shouldUseDarkTheme -> AppCompatDelegate.MODE_NIGHT_YES
+                settings.shouldFollowDeviceTheme -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+                else -> AppCompatDelegate.MODE_NIGHT_UNSPECIFIED
+            }
```

Comment:
@Amejia481 I noticed that the theme selection preference in the `Customize` settings page was ending up with both "Follow device theme" and either "Light" or "Dark" selected. I assume we only want one of those to ever be selected, and this change also allows us to fall back to the user's original preference. 

The `updateWallpaper` function is called when the `HomeFragment` creates its view anyway, so the theme will always match the wallpaper.

Reply:
Thank for finding the issue.  I'm able to reproduce, it looks like at some point we marked on both settings as `true`? Could you help me to clarify how not persisting the setting will address the issue? Are you not going to get flashes between themes if we don't persist?

Reply:
If we add `settings.shouldFollowDeviceTheme = false` shold fix the issue

Reply:
Is there any specific reason why we don't want to persist the theme adjustment ?

Reply:
This approach address the problem, but without persisting the theme setting we will be switching between themes  when the user starts the app, in some cases the user doesn't go directly to home,  could go directly browser screen when it has an opened tab.

Reply:
I think there is still some value in keeping in sync the setting and the theme. In some cases, as we  are adjusting the theme but no there setting, users could be in some inconsistent setting states, where the customization page says the selected theme is light but the applied one is the dark theme, because of the wallpaper. 
![image](https://user-images.githubusercontent.com/773158/148994890-06bd2b51-aa86-476c-9d1b-e988fb1b5a5b.png)

Reply:
@Amejia481 That's a good point about the home screen not always being the first view that is loaded, which could cause the theme to be out of sync.

One benefit of this implementation is that the theme is still in sync with what the user has selected. It allows us to return to the theme they had previously selected when they switch back to the default wallpaper. 

We could address your concern and keep this behavior by setting the theme in `FenixApplication` based on whether a wallpaper was selected or not. What do you think of that approach? If we go that way, should we do it in a follow-up PR?

Reply:
I think we could keep the actual behavior and improve it if need it in a follow up 👍🏽 .